src/AutoSaveStatus/AutoSaveStatusProvider.tsx

```diff
@@ -1,26 +1,27 @@
 import React, { PropsWithChildren, useCallback, useEffect, useMemo, useRef, useState } from "react";
 
 export enum AutoSaveStatus {
+  /** No calls are in-flight or just-recently-saved. */
   IDLE = "idle",
+  /** A call is actively in-flight. */
   SAVING = "saving",
+  /** A call is no longer actively in-flight, but has recently finished/can show confirmed. */
   DONE = "done",
+  /** A call is no longer actively in-fight, but has errored out. */
   ERROR = "error",
 }
 
 export interface AutoSaveStatusContextType {
   status: AutoSaveStatus;
-  /** Resets status to IDLE, particularly useful if "Error" or "Done" is stale */
-  resetStatus: VoidFunction;
```

Comment:
I removed `resetStatus` with the idea that it's the responsibility of the provider itself to manage when to reset/not-reset status, based on components notifying of their own/individual status.

Reply:
It already did do that, to an extent.

`resetStatus()` was a way for components to manually clear our the save status. i.e. if something failed, the user started typing again, and we wanted to show Idle/Ready instead of Error before actually firing a new save.